rayon-core/src/configuration.rs

```diff
@@ -55,19 +55,30 @@ pub struct Configuration {
 
     /// Closure to compute the name of a thread.
     get_thread_name: Option<Box<FnMut(usize) -> String>>,
+
+    /// Closure invoked on worker thread start.
+    start_handler: Option<StartHandler>,
+
+    /// Closure invoked on worker thread exit.
+    exit_handler: Option<ExitHandler>,
 }
 
 /// The type for a panic handling closure. Note that this same closure
 /// may be invoked multiple times in parallel.
 pub type PanicHandler = Arc<Fn(Box<Any + Send>) + Send + Sync>;
 
+pub type StartHandler = Arc<Fn() + Send + Sync>;
```

Comment:
I wonder if we should pass in the index of the thread, as we do for `get_thread_name`. Seems like probably yes.